src/app/core/public/api/impl/Circuit.ts

```diff
@@ -113,11 +115,26 @@ export abstract class CircuitImpl<
     }
 
     // Queries
-    public pickObjectAt(pt: Vector, space: Vector.Spaces = "world"): ComponentT | WireT | PortT | undefined {
+    public pickObjAt(pt: Vector, space: Vector.Spaces = "world"): ComponentT | WireT | PortT | undefined {
         const pos = ((space === "world") ? pt : this.view.toWorldPos(pt));
         return this.view.findNearestObj(pos).map((id) => this.getObj(id)).asUnion();
     }
-    public pickObjectRange(bounds: Rect): Array<ComponentT | WireT | PortT> {
+    public pickComponentAt(pt: Vector, space: Vector.Spaces = "world"): ComponentT | undefined {
```

Comment:
Could you try to factor out the common structure from these 3 functions (and maybe even the `pickObjAt`)?  NBD if that shows up in a later PR in this chain of PRs.

Reply:
I attempted -- can you take another look?

Reply:
Definitely better!